yang-lsp/io.typefox.yang/src/main/java/io/typefox/yang/validation/YangValidator.xtend

```diff
@@ -334,6 +370,63 @@ class YangValidator extends AbstractYangValidator {
 		];
 	}
 
+	@Check
+	def checkAugment(Augment it) {
+		// https://tools.ietf.org/html/rfc7950#section-7.17
+		// https://github.com/yang-tools/yang-lsp/issues/25
+		//
+		// (1) The target node MUST be either a container, list, choice, case, input,
+		// output, or notification node.
+		// (2) If the target node is a container, list, case, input, output, or
+		// notification node, the "container", "leaf", "list", "leaf-list",
+		// "uses", and "choice" statements can be used within the "augment" statement.
+		// (3) If the target node is a container or list node, the "action" and
+		// "notification" statements can be used within the "augment" statement.
+		// (4) If the target node is a choice node, the "case" statement or a
+		// shorthand "case" statement (see Section 7.9.2) can be used within the "augment" statement.
+		val target = path?.schemaNode;
+		if (target !== null) {
+			val validSubstatements = validAugmentStatements.get(target.eClass);
+			if (validSubstatements.nullOrEmpty) {
+				val EClass[] inputOrOutput = newArrayOfSize(2);
+				// Implicit `input` and `output` is added to all "rpc" and "action" statements. (See: RFC 7950 7.14)
+				// ScopeContextProvider.getQualifiedName(SchemaNode, QualifiedName, IScopeContext)
+				if (target.eClass === RPC || target.eClass === ACTION) {
```

Comment:
Actually it looks like you don't need that variable at all, since you could just issue the message, once you figured out the last segment is neither `input` nor `output`.